tags/annotate_views.py

```diff
@@ -16,17 +16,48 @@
 
 from core.decorators import block_POST_for_incompetent
 from legacy.models import Series, Sample
-from tags.models import Tag, SeriesTag, SampleTag
+from starapi.serializers import SampleAnnotationValidator
+from tags.models import Tag, SeriesTag
 from .models import ValidationJob, SerieValidation, SampleValidation, SerieAnnotation
 from .tasks import validation_workflow, calc_validation_stats
+from .misc import save_annotation, save_validation
 
 
 @login_required
 @block_POST_for_incompetent
 @render_to('tags/annotate.j2')
 def annotate(request):
     if request.method == 'POST':
-        return save_annotation(request)
+        user_id = request.user.id
+
+        raw_data = {
+            'series': request.POST['series_id'],
+            'tag': request.POST['tag_id'],
+            'column': request.POST['column'],
+            'regex': request.POST['regex'],
+            'values': dict(json.loads(request.POST['values'])),
+        }
+
+        serializer = SampleAnnotationValidator(data=raw_data)
+        if not serializer.is_valid():
+            messages.error(request, serializer.errors)
+            return redirect(request.get_full_path())
+
+        data = serializer.validated_data
+
+        with transaction.atomic():
+            old_annotation = first(SeriesTag.objects.filter(series=data['series'], tag=data['tag']))
```

Comment:
- use `QuerySet.first()` and `.order_by('id')`.

Note that while we allow tag several platforms at once in UI, I don't want to allow that in API.

Also this code should be in `save_annotation()` not duplicated here.